From ef5c521d58d2c9dadff618c4a6015eeaa19b429c Mon Sep 17 00:00:00 2001
From: Thomas Potthast <tpotthast@ti.com>
Date: Wed, 10 Aug 2011 16:12:56 -0500
Subject: [PATCH] Browser: Added an option to force a new main window

* Used by matrix-gui-v2 to ensure that new browser tabs are not
  opened in a -nogui window
* allows multiple main windows, which were previously not possible.
* Usage ./browser [-nogui] [-new] [url]

Signed-off-by: Thomas Potthast <tpotthast@ti.com>
---
 demos/browser/browserapplication.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/demos/browser/browserapplication.cpp b/demos/browser/browserapplication.cpp
index e2b68a6..c941006 100644
--- a/demos/browser/browserapplication.cpp
+++ b/demos/browser/browserapplication.cpp
@@ -211,9 +211,14 @@ void BrowserApplication::postLaunch()

     loadSettings();

+    QStringList args = QCoreApplication::arguments();
+    //add in option to force a new main window
+    if(args.contains("-new")){
+        this->newMainWindow();
+    }
+
     // newMainWindow() needs to be called in main() for this to happen
     if (m_mainWindows.count() > 0) {
-        QStringList args = QCoreApplication::arguments();
	if(args.contains("-nogui")){
            mainWindow()->kioskMode();
            if(args.count() > 1){
--
1.7.0.4
